tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedServer.java

```diff
@@ -281,9 +290,18 @@ private static List<Module> chooseMailbox(MailboxConfiguration mailboxConfigurat
         return ImmutableList.of();
     }
 
+    private static class FirebaseListenerDistributedModule extends AbstractModule {
+        @ProvidesIntoSet
+        InitializationOperation registerFirebaseListener(@Named(InjectionKeys.JMAP) RabbitMQEventBus instance, FirebasePushListener firebasePushListener) {
+            return InitilizationOperationBuilder
+                .forClass(FirebasePushListenerRegister.class)
+                .init(() -> instance.register(firebasePushListener));
+        }
+    }
+
     private static List<Module> chooseFirebase(FirebaseModuleChooserConfiguration moduleChooserConfiguration) {
         if (moduleChooserConfiguration.enable()) {
-            return List.of(new CassandraFirebaseSubscriptionRepositoryModule(), new FirebaseCommonModule());
+            return List.of(new CassandraFirebaseSubscriptionRepositoryModule(), new FirebaseCommonModule(), new FirebaseListenerDistributedModule());
```

Comment:
Why don't we put the listener module inside in FirebaseCommonModule?
Listener is a part of common

Reply:
see  `PushListener` for the same. I would happy to get an explanation why you did that also :p

Reply:
just checked, because `RabbitMQEventBus`